cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i

```diff
@@ -334,39 +334,37 @@ Status KNNClassificationTrainBatchKernel<algorithmFpType, training::defaultDense
 
         daal::threader_for(blockCount, blockCount, [=, &bboxTLS](int iBlock) {
             BBox * const bboxLocal = bboxTLS.local();
-            if (bboxLocal)
-            {
-                const size_t first = iBlock * rowsPerBlock;
-                const size_t last  = min<cpu>(static_cast<decltype(xRowCount)>(first + rowsPerBlock), xRowCount);
 
-                if (first < last)
-                {
-                    BBox b;
-                    size_t i = first;
-                    b.upper  = dx[indexes[i]];
-                    b.lower  = dx[indexes[i]];
-                    PRAGMA_IVDEP
-                    for (++i; i < last; ++i)
-                    {
-                        if (b.lower > dx[indexes[i]])
-                        {
-                            b.lower = dx[indexes[i]];
-                        }
-                        if (b.upper < dx[indexes[i]])
-                        {
-                            b.upper = dx[indexes[i]];
-                        }
-                    }
+            const size_t first = iBlock * rowsPerBlock;
```

Comment:
Please add error handling. The check on the line #333 is probably useless as the lambda at the line #319 is called only when the .local() method is invoked inside daal::threader_for.
```suggestion
            DAAL_CHECK_MALLOC_THR(bboxLocal);
            const size_t first = iBlock * rowsPerBlock;
```

Reply:
resolved